### PR TITLE
Upgrade dbt-utils

### PR DESCRIPTION
**Pull request by mdesmet**

### CODE REVIEW

**Comment by hovaesco** [comment]
How about https://github.com/starburstdata/dbt-trino-utils/pull/4? Should it be closed?

File: docker/dbt/Dockerfile [hunk 1]
```diff
@@ -11,8 +11,9 @@ RUN apt-get update && \
     apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/
 RUN pip3 install --upgrade pip
 RUN mkdir ${DBT_DIR}
-# TODO: change with release version
-RUN pip install git+https://github.com/starburstdata/dbt-trino.git
+# TODO: change with release version once PR is merged
+RUN pip3 install git+https://github.com/mdesmet/dbt-trino.git@feature/get_colums_in_relation
```
Comment (mdesmet) [inline docker/dbt/Dockerfile]: This should be changed after we merge that branch on dbt-trino

File: macros/dbt_utils/sql/date_spine.sql [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    select *
+    from all_periods
+    where date_{{datepart}} <= cast(cast({{ end_date }} as date) as timestamp)
+
+)
+
+select * from filtered
+
+{% endmacro %}
```
Comment (hovaesco) [inline macros/dbt_utils/sql/date_spine.sql]: nit: new line here and in other places

File: macros/dbt_utils/sql/date_spine.sql [hunk 3]
```diff
@@ -0,0 +1,46 @@
+{% macro trino__date_spine(datepart, start_date, end_date) %}
+
+
+{# call as follows:
```
Comment (hovaesco) [inline macros/dbt_utils/sql/date_spine.sql:4]: Is the comment needed?

File: macros/dbt_utils/sql/deduplicate.sql [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{%- macro trino__deduplicate(relation, partition_by, order_by) -%}
+    {#
+    -- This is a temporary solution until https://github.com/trinodb/trino/issues/14455 is resolved
+    -- It leaks the rn column compared to the NATURAL JOIN solution from dbt
```
Comment (hovaesco) [inline macros/dbt_utils/sql/deduplicate.sql:4]: What it means `rn`?

**Review by hovaesco — APPROVED** [review]

File: macros/dbt_utils/sql/deduplicate.sql (hunk 4 above)
Comment (mdesmet) [reply macros/dbt_utils/sql/deduplicate.sql:4]: So the NATURAL JOIN implementation (which is not supported) would not leak the row number column as output of the deduplicate query.

File: macros/dbt_utils/sql/date_spine.sql (hunk 3 above)
Comment (mdesmet) [reply macros/dbt_utils/sql/date_spine.sql:4]: it comes from the original dbt implementation, I would say it's ok to keep it.

**Pull request merged**